str_chr.c.gnu

```diff
@@ -0,0 +1,9 @@
+#define _GNU_SOURCE
```

Comment:
Testing on musl (`Linux t470s.z0.is 5.4.35-0-lts #1-Alpine SMP Thu, 23 Apr 2020 10:36:12 UTC x86_64 Linux`):

```
t470s:$ cp str_chr.c.gnu str_chr.c && make clean test 2>&1 | grep -m 1 '^real'
real    0m 2.32s
t470s:$ cp str_chr.c.impl str_chr.c && make clean test 2>&1 | grep -m 1 '^real'
real    0m 2.32s
t470s:$ cp str_chr.c.orig str_chr.c && make clean test 2>&1 | grep -m 1 '^real'
real    0m 2.33s
```

It did not look like the performance improvement was worth the extra check, but maybe the gap is bigger on other libcs.

[EDIT]: added the original implementation

Reply:
Here is how I tested:

- Added `for (int i = 0; i < 200000; i++) { [...] }` around `ck_assert_uint_eq(str_rchr([...]))`
- Added `time` before `./$$tbin || exit 1 ;`
- Ran the above commands.

Reply:
The main idea was not to get a performance improvement, but just to junk that code. Nice if it doesn't hurt anyway ;)